connector/connector.go

```diff
@@ -437,6 +439,39 @@ func (f factory) LogsToLogsStability() component.StabilityLevel {
 	return f.logsToLogsStabilityLevel
 }
 
+func (f factory) Stability(expType, recType component.Type) component.StabilityLevel {
+	switch expType {
+	case component.DataTypeTraces:
+		switch recType {
+		case component.DataTypeTraces:
+			return f.tracesToTracesStabilityLevel
+		case component.DataTypeMetrics:
+			return f.tracesToMetricsStabilityLevel
+		case component.DataTypeLogs:
+			return f.tracesToLogsStabilityLevel
+		}
+	case component.DataTypeMetrics:
+		switch recType {
+		case component.DataTypeTraces:
+			return f.metricsToTracesStabilityLevel
+		case component.DataTypeMetrics:
+			return f.metricsToMetricsStabilityLevel
+		case component.DataTypeLogs:
+			return f.metricsToLogsStabilityLevel
+		}
+	case component.DataTypeLogs:
+		switch recType {
+		case component.DataTypeTraces:
+			return f.logsToTracesStabilityLevel
+		case component.DataTypeMetrics:
+			return f.logsToMetricsStabilityLevel
+		case component.DataTypeLogs:
+			return f.logsToLogsStabilityLevel
+		}
+	}
+	return component.StabilityLevelUndefined
+}
```

Comment:
This code doesn't strictly need to be part of the API, but it felt silly to reimplement it elsewhere.